### ISSUE TITLE
net: no deadline support for net.Dial*, Resolve*, Lookup*, etc

### ISSUE THREAD

**Issue by niemeyer** [golang/go#6340]
<pre>The deadline support introduced in the net package is currently restricted to dialing
(network, address string) via net.DialTimeout or Dialer, or to already established
connections. It would be valuable to have more functionality covered.

The specific use case at hand when this was missed was with ResolveTCPAddr and DialTCP.
It's not possible to timeout the resolution, and once an address is obtained, the only
way to dial with a timeout is by stringifying the address again and submitting into
DialTimeout or Dialer for re-parsing.</pre>

**Comment by LionNatsu** [comment]
Maybe close this?

**Comment by bradfitz** [comment]
Thanks. Yeah, this was fixed in Go 1.8:

```
pkg net, method (*Resolver) LookupAddr(context.Context, string) ([]string, error)
pkg net, method (*Resolver) LookupCNAME(context.Context, string) (string, error)
pkg net, method (*Resolver) LookupHost(context.Context, string) ([]string, error)
pkg net, method (*Resolver) LookupIPAddr(context.Context, string) ([]IPAddr, error)
pkg net, method (*Resolver) LookupMX(context.Context, string) ([]*MX, error)
pkg net, method (*Resolver) LookupNS(context.Context, string) ([]*NS, error)
pkg net, method (*Resolver) LookupPort(context.Context, string, string) (int, error)
pkg net, method (*Resolver) LookupSRV(context.Context, string, string, string) (string, []*SRV, error)
pkg net, method (*Resolver) LookupTXT(context.Context, string) ([]string, error)
pkg net, type Dialer struct, Resolver *Resolver
pkg net, type Resolver struct
pkg net, type Resolver struct, PreferGo bool
pkg net, var DefaultResolver *Resolver
```
